src/super_gradients/training/models/segmentation_models/segformer.py

```diff
@@ -372,12 +386,58 @@ def _remove_auxiliary_heads(self):
     def replace_head(self, new_num_classes: int, new_decoder_embed_dim: int):
         self.decode_head = SegFormerHead(encoder_dims=self.encoder_embed_dims, embed_dim=new_decoder_embed_dim, num_classes=new_num_classes)
 
-    def forward(self, x: torch.Tensor) -> torch.Tensor:
+    def _forward(self, x: torch.Tensor) -> torch.Tensor:
         features = self._backbone(x)
         out = self.decode_head(features)
         out = F.interpolate(out, size=x.shape[2:], mode="bilinear", align_corners=False)
         return out
 
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        if self.is_SWI:
+            return self.sliding_window(x, self.num_classes, self.sliding_window_stride, self.sliding_window_size)
+        else:
+            return self._forward(x)
+
+    def sliding_window(self, img: torch.Tensor, num_classes: int, stride: Tuple[int, int], crop_size: Tuple[int, int]) -> torch.Tensor:
+        """
+        Inference by sliding-window with overlap.
+
+        If h_crop > h_img or w_crop > w_img, the small patch will be used to
+        decode without padding.
+        """
```

Comment:
If you can add params description (:param <name>: <desc>)

Reply:
Also, make sure to adhere to PEP docs format (add Pycharm plugin for convinience).

Reply:
Adding to @Louis-Dupont , it would be great to give some example of what goes down under the hood when we use sliding window inference.

Reply:
Fixed on c79d0f2